src/lib/support/PersistedCounter.h

```diff
@@ -104,12 +104,22 @@ class PersistedCounter : public MonotonicallyIncreasingCounter<T>
         }
 #endif
 
-        ReturnErrorOnFailure(PersistNextEpochStart(startValue + aEpoch));
+        ReturnErrorOnFailure(PersistNextEpochStart(static_cast<T>(startValue + aEpoch)));
 
         // This will set the starting value, after which we're ready.
         return MonotonicallyIncreasingCounter<T>::Init(startValue);
     }
 
+    CHIP_ERROR AdvanceBy(T value) override
+    {
+        VerifyOrReturnError(mStorage != nullptr, CHIP_ERROR_INCORRECT_STATE);
+        VerifyOrReturnError(mKey.IsInitialized(), CHIP_ERROR_INCORRECT_STATE);
+
+        ReturnErrorOnFailure(MonotonicallyIncreasingCounter<T>::AdvanceBy(value));
+
+        return VerifyAndPersistNextEpochStart(MonotonicallyIncreasingCounter<T>::GetValue());
```

Comment:
This could use a comment explaining why we are using current value as the reference epoch here.

Reply:
done